modules/openapi-generator/src/main/resources/typescript-fetch/apis.mustache

```diff
@@ -124,7 +124,7 @@ export class {{classname}} extends runtime.BaseAPI {
             {{/isDateTime}}
             {{^isDateTime}}
             {{#isDate}}
-            queryParameters['{{baseName}}'] = (requestParameters.{{paramName}} as any).toISOString().substr(0,10);
+            queryParameters['{{baseName}}'] = new Date((requestParameters.{{paramName}} as Date).getTime() - (requestParameters.{{paramName}} as Date).getTimezoneOffset()*60000).toISOString().substr(0,10);
```

Comment:
can you clarify how this resolves the issue?
maybe this could be extracted to a helper function

Reply:
Typically, when creating a Javascript Date object for a "date" the time 00:00 is used. If you do this using the local timezone or no timezone (then the local timezone is used), then you will face this:

- `new Date('2022-06-05T00:00:00').toISOString()`  -> '2022-**06-04**T22:00:00.000Z'
- `new Date('2022-06-05T00:00:00+0200').toISOString()`  -> '2022-**06-04**T22:00:00.000Z'

This is the way most of the frameworks like date-pickers retrieve the Date object and respect to the user's timezone. Since the `toISOString` method is used to serialize the Javascript Date as a "date" format, the date serializing into JSON is changing.

To fix this, a new Date is created substracting the local timezone's offset. This will give 00:00:00 at UTC and therefore `toISOString` will return the desired date.

Even if the Date is created using this

- `new Date('2022-06-05').toISOString()`  -> '2022-**06-05**T00:00:00.000Z'

which gives the right UTC date, the proposed timezone-offset fix will change work because in this situation it does not change anything:

`new Date(new Date('2022-06-05').getTime() - new Date('2022-06-05').getTimezoneOffset()).toISOString()`
   -> ''2022-**06-05**T00:00:00.120Z''

By-the-way: It might be a good idea to move this to a helper/utility since other Typescript/Javascript template might face the same problem. I'm not familiar with the project and cannot propose this in a proper way and also have not the resources to test other templates as well. I'm in a project which should be done since August which means I hope you will accept this PR anyway. Someone, who is able to, can do another PR for the utility and the rest of the Typescript/Javascript templates.

Thx, Stephan

Reply:
This approach just leads to a new set of bugs. `.getTimezoneOffset()` will give you the offset of the *system* timezone, which may well not be the timezone that you're trying to work with. 

As a real world example, as a developer working in Europe/London, building a web app used by companies in America/New_York, I might create a `Date` that refers to the correct point in time for that company. But the `getTimezoneOffset` method of the Date object will show an offset for London, *not* New_York. So the output sent to the API would still be incorrect. 

As we don't yet have `PlainDate` in JS, I think we might need to make do with supplying strings for the time being...